bubble_sort.go

```diff
@@ -20,7 +20,7 @@ func bubbleSort2(arr []int) []int {
 	for i := 0; i < len(arr); i++ {
 		done = true
 		for j := 0; j < len(arr)-i-1; j++ {
-			if arr[j] > arr[j+1] {
+			if arr[j] > arr[j+2] {
```

Comment:
Runtime Error Check: The current change could lead to an index out of range error if `j+2` exceeds the length of the array. Ensure that the loop boundaries are correctly set to prevent such errors.